src/api/pickAppResource.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem, ITreeItemPickerContext } from "@microsoft/vscode-azext-utils";
+import { PickAppResourceOptions } from "@microsoft/vscode-azext-utils/hostapi";
+import { ext } from "../extensionVariables";
+import { SubscriptionTreeItem } from "../tree/SubscriptionTreeItem";
+
+export async function pickAppResource<T extends AzExtTreeItem>(context: ITreeItemPickerContext, options?: PickAppResourceOptions): Promise<T> {
+    const subscription: SubscriptionTreeItem = await ext.appResourceTree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
+    const appResource = await subscription.pickAppResource(context, options);
+
+    if (!appResource.resolveResult) {
+        await appResource.resolve(true, context);
```

Comment:
Do we need to await the resolve here? Unless the options have `expectedChildContextValue`, we should be able to just return the `proxyTreeItem`.  

Haven't tested this, but could there be a noticeable delay if the resource isn't resolved?  Also, if the command we call is from Resource Groups (`openInPortal`) we'll have to wait for it to activate the extension and resolve as well.

I imagine that the extension will have to be activated since we're resolving the resource even if unneeded.  However, if it's that extensions registered command, it'll have to activate and should resolve all the resolvables.

Reply:
Would you expect some commands to await the resolve after calling pickAppResource?